[DDP] Remove SPMD from get_bucket_tensors

rohan-varma opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #61020
* #61019
* #61018
* **#61017**

Removes SPMD nested vector logic from this codepath. This is mostly in preparation for the next diffs in this stack which enable support for join with comm. hook.

Differential Revision: [D29477360](https://our.internmc.facebook.com/intern/diff/D29477360/)

rohan-varma committed:
[DDP] Remove SPMD from get_bucket_tensors

Removes SPMD nested vector logic from this codepath. This is mostly in preparation for the next diffs in this stack which enable support for join with comm. hook.

Differential Revision: [D29477360](https://our.internmc.facebook.com/intern/diff/D29477360/)

[ghstack-poisoned]

wayi1 reviewed (approved):
Thanks for the refactoring!

rohan-varma committed:
Update on "[DDP] Remove SPMD from get_bucket_tensors"


Removes SPMD nested vector logic from this codepath. This is mostly in preparation for the next diffs in this stack which enable support for join with comm. hook.

Differential Revision: [D29477360](https://our.internmc.facebook.com/intern/diff/D29477360/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[DDP] Remove SPMD from get_bucket_tensors"


Removes SPMD nested vector logic from this codepath. This is mostly in preparation for the next diffs in this stack which enable support for join with comm. hook.

Differential Revision: [D29477360](https://our.internmc.facebook.com/intern/diff/D29477360/)

[ghstack-poisoned]
